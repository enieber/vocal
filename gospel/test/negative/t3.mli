(**************************************************************************)
(*                                                                        *)
(*  GOSPEL -- A Specification Language for OCaml                          *)
(*                                                                        *)
(*  Copyright (c) 2018- The VOCaL Project                                 *)
(*                                                                        *)
(*  This software is free software, distributed under the MIT license     *)
(*  (as described in file LICENSE enclosed).                              *)
(**************************************************************************)

(*@ function f (x:float): bool = x = 2 *)

(* ERROR Type mismatch float and integer *)
