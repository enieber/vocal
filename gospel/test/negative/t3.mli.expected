
*******************************
********** Parsed file ********
*******************************
[@@@gospel  function f (x:float): bool = x = 2 ]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Gospelstdlib *)

(*@ function f  ... *)
File "t3.mli", line 11, characters 33-34:
Error: Type mysmatch. Cannot match float with integer
